devito/ir/ietxdsl/cluster_to_ssa.py

```diff
@@ -120,41 +118,45 @@ def _convert_eq(self, eq: LoweredEq):
         # Get the stepping dimension. It's usually time, and usually the first one.
         # Getting it here; more readable and less input assumptions :)
         step_dim = grid.stepping_dim
-
-        # Get all functions used in the equation
-        functions = OrderedSet(*(f.function for f in retrieve_function_carriers(eq)))
         # We identify time buffers by their function and positive time offset.
         # e.g., u(t+2, x-1, y) would be indentified as (u, 2)
         # NB. We map time offsets to positive with simple modulo arithmetic: if u has a
         # a time size of 3, then u(t-2, ...) is identified as u(t, 1).
 
         # We store a list of those here to help the following steps.
-        self.time_buffers = [(f, i) for f in functions for i in range(f.time_size)]
 
         # Also, store the output function's time buffer of this equation
         output_time_offset = (eq.lhs.indices[step_dim] - step_dim) % eq.lhs.function.time_size
         self.out_time_buffer = (output_function, output_time_offset)
 
-        # For each used time_buffer, define a stencil.field type for the function .
-        fields_types = [field_from_function(f) for (f, _) in self.time_buffers]
+        # Bounds and step boilerpalte
+        lb = iet_ssa.LoadSymbolic.get(step_dim.symbolic_min._C_name, builtin.IndexType())
+        ub = iet_ssa.LoadSymbolic.get(step_dim.symbolic_max._C_name, builtin.IndexType())
+        one = arith.Constant.from_int_and_width(1, builtin.IndexType())
+        try:
+            step = arith.Constant.from_int_and_width(
```

Comment:
Actually l.135 is uselss

Reply:
Idem, not useless, just used to defined time_M+1
There are probably elegant alternatives, but I'd say it's no priority at this point?